### ISSUE TITLE
Serious performance issues

### ISSUE THREAD

**Issue by alkalimc** [Hexnet111/SuitSaver#2]
**Hello, this is a player feedback from China, I use translation software to provide descriptions, so if there is any semantic discrepancy, please be considerate.**
I saw a significant performance penalty while working on a large modpack, going from about 240 fps to 80 fps and a significant drop in GPU usage, so I judged this to be an unusual issue, and after troubleshooting, I determined that the problem was with the plugin. When all mods in the modpack (which contain about 100 plugins) are enabled, the frame rate decreases to the point where it is not playable, especially when the player is the host, from about 80 fps to 40 fps, and the 1% low fps is very low (about 10 fps) and does not work properly, but I think the functionality of this add-on is very important, so I wanted you to report this issue and hope you will try to fix it. I'm still not sure if it's an issue with the mod itself or if it's incompatible with other mods, but I can provide you with a list of my mods later if you need to.

**Comment by Hexnet111** [comment]
> **Hello, this is a player feedback from China, I use translation software to provide descriptions, so if there is any semantic discrepancy, please be considerate.** I saw a significant performance penalty while working on a large modpack, going from about 240 fps to 80 fps and a significant drop in GPU usage, so I judged this to be an unusual issue, and after troubleshooting, I determined that the problem was with the plugin. When all mods in the modpack (which contain about 100 plugins) are enabled, the frame rate decreases to the point where it is not playable, especially when the player is the host, from about 80 fps to 40 fps, and the 1% low fps is very low (about 10 fps) and does not work properly, but I think the functionality of this add-on is very important, so I wanted you to report this issue and hope you will try to fix it. I'm still not sure if it's an issue with the mod itself or if it's incompatible with other mods, but I can provide you with a list of my mods later if you need to.

Hello,

Thank you for letting me know of this issue. I had a quick look at the source code and I'm unsure as to what could cause this. This may be due to a compatibility issue with another mod. Could you perhaps send me your mod list?

Thank you in advance.

**Comment by alkalimc** [comment]
Sorry for taking so long to get back to you, but before that I'm ruling out the possibility of a performance penalty from a 13th Gen Intel CPU thread scheduling error, which manifests as the CPU efficiency core workload is too high and the GPU workload is too low. After I forced the LC to work on the performance core I ruled out the above possibility, I'll give you a list of mods and a comparison of the fps before and after turning your mods on.
# Before
[![21010bf989af5e74c6e98458b15427f6.png](https://s1.imagehub.cc/images/2024/07/24/21010bf989af5e74c6e98458b15427f6.png)](https://www.imagehub.cc/image/%E5%B1%8F%E5%B9%95%E6%88%AA%E5%9B%BE-2024-07-24-181143.bsTGuU)
# After
[![d8a14018498942afb1a1c84962a3e9d0.png](https://s1.imagehub.cc/images/2024/07/24/d8a14018498942afb1a1c84962a3e9d0.png)](https://www.imagehub.cc/image/%E5%B1%8F%E5%B9%95%E6%88%AA%E5%9B%BE-2024-07-24-180945.bsTE4q)
# List
[![dd28975451127497ca6b56b455ad9731.png](https://s1.imagehub.cc/images/2024/07/24/dd28975451127497ca6b56b455ad9731.png)](https://www.imagehub.cc/image/%E5%B1%8F%E5%B9%95%E6%88%AA%E5%9B%BE-2024-07-24-181201.bsT4bB)
[![e73199e52f4fc31b7c9732e55e452758.png](https://s1.imagehub.cc/images/2024/07/24/e73199e52f4fc31b7c9732e55e452758.png)](https://www.imagehub.cc/image/%E5%B1%8F%E5%B9%95%E6%88%AA%E5%9B%BE-2024-07-24-181230.bsTPS0)
[![bb1b4cf1b2b84f18affea0ceca449ea4.png](https://s1.imagehub.cc/images/2024/07/24/bb1b4cf1b2b84f18affea0ceca449ea4.png)](https://www.imagehub.cc/image/%E5%B1%8F%E5%B9%95%E6%88%AA%E5%9B%BE-2024-07-24-181249.bsTcBz)
In addition, I can send you the current modpack directly, so please contact me if you need it.

**Comment by alkalimc** [comment]
To add, my hardware environment:
- 13980HX
- 64GB RAM
- 4070laptop

**Comment by Hexnet111** [comment]
> I can send you the current modpack directly, so please contact me if you need it.

Hello,

Yes, it would help me a bunch if you could send me the modpack directly. Thank you for providing further details about this. I'm still not sure what could be causing the issue.

**Comment by alkalimc** [comment]
Okay, since it's too late, probably 3 am China time (UTC+8), I'll try to get back to you by 11 am tomorrow morning China time, although I don't know your time zone there, but you can calculate the time 8 hours later in your region by yourself.
In the meantime, I found some other mods that exhibit the same problem, including but not limited to [GeneralImprovements](https://thunderstore.io/c/lethal-company/p/ShaosilGaming/GeneralImprovements/) and [LC Simplified Chinese Localization](https://thunderstore.io/c/lethal-company/p/NakiriFox/LC_Simplified_Chinese_Localization/) I can't determine the association between them, but provide them to you as information.
Your mod functionality is very useful and important, and I sincerely hope that you will solve this problem.
Can you suggest me a website where I can upload my modpacks , or can I create a new repository for you to download?

**Comment by Hexnet111** [comment]
> Okay, since it's too late, probably 3 am China time (UTC+8), I'll try to get back to you by 11 am tomorrow morning China time, although I don't know your time zone there, but you can calculate the time 8 hours later in your region by yourself. In the meantime, I found some other mods that exhibit the same problem, including but not limited to [GeneralImprovements](https://thunderstore.io/c/lethal-company/p/ShaosilGaming/GeneralImprovements/) and [LC Simplified Chinese Localization](https://thunderstore.io/c/lethal-company/p/NakiriFox/LC_Simplified_Chinese_Localization/) I can't determine the association between them, but provide them to you as information. Your mod functionality is very useful and important, and I sincerely hope that you will solve this problem. Can you suggest me a website where I can upload my modpacks , or can I create a new repository for you to download?

Hello,

You should be able to export your modpack through code. Go to the settings, then head over to the profile section. After that, scroll down until you find "export profile as code", and send me the code here.

**Comment by alkalimc** [comment]
BepInEx-BepInExPack-5.4.2100
Sligili-HDLethalCompany-1.5.6
AinaVT-LethalConfig-1.4.2
x753-More_Suits-1.4.3
Verity-TooManySuits-1.1.0
BunyaPineTree-ModelReplacementAPI-2.4.7
MK_MODS-BlueArchivePlayerModelReplay-8.1.0
LuckyJesta-ArknightsPlayerModel-7.0.1
notnotnotswipez-MoreCompany-1.9.4
anormaltwig-LateCompany-1.0.16
FlipMods-ReservedItemSlotCore-2.0.31
FlipMods-ReservedFlashlightSlot-2.0.6
FlipMods-ReservedWalkieSlot-2.0.5
FlipMods-BetterStamina-1.3.2
FlipMods-FasterItemDropship-1.3.0
FlipMods-ObjectVolumeController-1.1.0
FlipMods-ReservedWeaponSlot-1.1.4
FlipMods-ReservedSprayPaintSlot-1.1.1
FlipMods-ReservedUtilitySlot-1.0.4
Rune580-LethalCompany_InputUtils-0.7.7
Renegades-FlashlightToggle-1.5.0
Renegades-WalkieUse-1.5.0
Lordfirespeed-OdinSerializer-2022.11.9
xilophor-StaticNetcodeLib-1.1.1
EliteMasterEric-Coroner-2.0.0
EliteMasterEric-CleanerLogs-1.0.0
EliteMasterEric-GarageDoorFix-1.0.0
EliteMasterEric-SlimeTamingFix-1.0.2
Hexnet111-SuitSaver-1.2.1

**Comment by alkalimc** [comment]
This is more or less the smaller package that shows the problem, comparing the difference in fps before and after Suitsaver is enabled

**Comment by Hexnet111** [comment]
> This is more or less the smaller package that shows the problem, comparing the difference in fps before and after Suitsaver is enabled

Hey there,

I've downloaded the mods you listed, and I cannot replicate the same exact problem. However, one thing I did notice, is that the suit models added by the "BlueArchivePlayerModelReplay" significantly reduce fps when equipped. Some when equipped only reduce by 30 - 40 fps, but some reduced by up to 80 fps. The behavior seems to be the same regardless if my mod is installed or not.

One thing I'd like to suggest is trying to maybe clear the config folder, and seeing whether or not that helps. Otherwise though, I don't think there's much I can do, as I can't replicate the same issue on my end.
